test/Dialect/SPIRV/canonicalize.mlir

```diff
@@ -1,5 +1,111 @@
 // RUN: mlir-opt %s -split-input-file -canonicalize | FileCheck %s
 
+//===----------------------------------------------------------------------===//
+// spv.AccsessChain
+//===----------------------------------------------------------------------===//
+
+func @access_chain_canonicalization_full_0() -> f32 {
+  // CHECK: %[[INDEX_0:.*]] = spv.constant 0 : i32
+  // CHECK-NEXT: %[[COMPOSITE:.*]] = spv.Variable : !spv.ptr<!spv.struct<!spv.array<4 x !spv.array<4 x f32>>, !spv.array<4 x i32>>, Function>
```

Comment:
We can simplify the `CHECK`s here. No need to check all the details. What about something like

```
// CHECK: %[[INDEX:.*]] = spv.constant 0
// CHECK: %[[VAR:.*]] = spv.Variable
// CHECK: %[[PTR:.*]] = spv.AccessChain %[[VAR]][%[[INDEX]], %[[INDEX]], %[[INDEX]]]
// CHECK: spv.Load "Function" %[[PTR]]
```
The type consistency is guaranteed by each op's validation itself and we should be fine to rely on that.

Similarly for all the test cases.